src/DynamoCore/Graph/Workspaces/WorkspaceModel.cs

```diff
@@ -565,6 +565,32 @@ private void OnSyncWithDefinitionEnd(NodeModel nodeModel)
         {
             hasNodeInSyncWithDefinition = false;
         }
+        public void DispatchOnUIThread(Action a, bool blocking = false)
+        {
+            OnDispatchedToUI(this, new UIDispatcherEventArgs(a,blocking));
+        }
+
+        private void OnDispatchedToUI(object sender, UIDispatcherEventArgs e)
```

Comment:
This is great, I remember there are some other places where I check if dispatcher is null, if yes, just execute it. This is worth making an APi

Reply:
So this method looks like it can do the following:
1. Execute code in the same thread context
  a. Async
  b. Sync
2. Execute code in another thread context
  a. Async
  b. Sync

Reply:
yep